index.js

```diff
@@ -48,8 +48,27 @@ const supportsAbortController = typeof globals.AbortController === 'function';
 const supportsStreams = typeof globals.ReadableStream === 'function';
 const supportsFormData = typeof globals.FormData === 'function';
 
+const mergeHeaders = (source1, source2) => {
+	const result = new globals.Headers();
+
+	for (const source of [source1, source2]) {
+		const isHeadersInstance = source instanceof globals.Headers;
+
+		for (const [key, value] of new globals.Headers(source)) {
```

Comment:
Define `new globals.Headers(source)` before the `for` loop. Two keywords near each other? It should be illegal :P

Alternatively you can do this:

```suggestion
		const iterator = isHeadersInstance ? headers.entries() : Object.entries(headers);

		for (const [key, value] of iterator) {
```

Reply:
I agree with the keywords :)
The iterator does not differ, either way, the source is now always a `Headers` instance.